reactor/src/main/java/com/hivemq/client/mqtt/mqtt5/reactor/Mqtt5ReactorClient.java

```diff
@@ -0,0 +1,323 @@
+     * subscribing (in terms of Reactive Streams) to the returned {@link Mono}.
+     *
+     * @param unsubscribe the Unsubscribe message sent to the broker during unsubscribe.
+     * @return the {@link Mono} which
+     *         <ul>
+     *         <li>succeeds with the UnsubAck message if at least one Topic Filter of the Unsubscribe message was
+     *         successfully unsubscribed (the UnsubAck message contains at least one Reason Code that is not an Error
+     *         Code),</li>
+     *         <li>errors with a {@link com.hivemq.client.mqtt.mqtt5.exceptions.Mqtt5UnsubAckException
+     *         Mqtt5UnsubAckException} wrapping the UnsubAck message if it only contains Error Codes or</li>
+     *         <li>errors with a different exception if an error occurred before the Unsubscribe message was sent or
+     *         before a UnsubAck message was received.</li>
+     *         </ul>
+     */
+    @NotNull Mono<Mqtt5UnsubAck> unsubscribe(@NotNull Mqtt5Unsubscribe unsubscribe);
+
+    /**
+     * Fluent counterpart of {@link #unsubscribe(Mqtt5Unsubscribe)}.
+     * <p>
+     * Calling {@link Mqtt5UnsubscribeBuilder.Nested.Complete#applyUnsubscribe()} on the returned builder has the same
+     * effect as calling {@link #unsubscribe(Mqtt5Unsubscribe)} with the result of {@link
+     * Mqtt5UnsubscribeBuilder.Complete#build()}.
+     *
+     * @return the fluent builder for the Unsubscribe message.
+     * @see #unsubscribe(Mqtt5Unsubscribe)
+     */
+    default @NotNull Mqtt5UnsubscribeBuilder.Nested.Start<Mono<Mqtt5UnsubAck>> unsubscribeWith() {
+        return new MqttUnsubscribeBuilder.Nested<>(this::unsubscribe);
+    }
+
+    /**
+     * Creates a {@link Flux} for publishing the Publish messages emitted by the given {@link Publisher}.
+     * <p>
+     * The returned {@link Flux} represents the source of {@link Mqtt5PublishResult}s each corresponding to a Publish
+     * message emitted by the given {@link Publisher}. Calling this method does not start publishing yet. Publishing is
+     * performed lazy and asynchronous. When subscribing (in terms of Reactive Streams) to the returned {@link Flux} the
+     * client subscribes (in terms of Reactive Streams) to the given {@link Publisher}.
+     *
+     * @param publisher the source of the Publish messages to publish.
+     * @return the {@link Flux} which
+     *         <ul>
+     *         <li>emits {@link Mqtt5PublishResult}s each corresponding to a Publish message,</li>
+     *         <li>completes when the given {@link Publisher} completes,</li>
+     *         <li>errors with the same exception when the given {@link Publisher} errors or</li>
+     *         <li>errors with a different exception if an error occurred before all Publish messages of the given
+     *         {@link Publisher} are answered by a {@link Mqtt5PublishResult} (e.g. {@link
+     *         com.hivemq.client.mqtt.exceptions.MqttSessionExpiredException MqttSessionExpiredException}).</li>
```

Comment:
This is wrong since 1.1, only individual PublishResults contain an error, javadoc for RxClient must also be updated